Test failure on is_not_null().any() (main branch)

https://github.com/pola-rs/polars/actions/runs/25729935929/job/75552407139?pr=27594

```
______________________ test_is_not_null_followed_by_any _______________________
[gw3] win32 -- Python 3.14.4 D:\a\polars\polars\py-polars\.venv\Scripts\python.exe
tests\unit\lazyframe\test_optimizations.py:89: in test_is_not_null_followed_by_any
    assert_frame_equal(expected_df, result_lf.collect())
src\polars\_utils\deprecation.py:128: in wrapper
    return function(*args, **kwargs)
src\polars\_utils\deprecation.py:128: in wrapper
    return function(*args, **kwargs)
src\polars\_utils\deprecation.py:128: in wrapper
    return function(*args, **kwargs)
E   AssertionError: DataFrames are different (value mismatch for column "val")
E   [left]: shape: (3,)
E   Series: 'val' [bool]
E   [
E   	true
E   	false
E   	true
E   ]
E   [right]: shape: (3,)
E   Series: 'val' [bool]
E   [
E   	true
E   	true
E   	true
E   ]
=========================== short test summary info ===========================
FAILED tests/unit/lazyframe/test_optimizations.py::test_is_not_null_followed_by_any - AssertionError: DataFrames are different (value mismatch for column "val")
[left]: shape: (3,)
Series: 'val' [bool]
[
	true
	false
	true
]
[right]: shape: (3,)
Series: 'val' [bool]
[
	true
	true
	true
]
==== 1 failed, 34267 passed, 141 skipped, 11 xfailed in 1317.01s (0:21:57) ====
```

--- a/crates/polars-expr/src/reduce/is_empty.rs
+++ b/crates/polars-expr/src/reduce/is_empty.rs
@@ -60,7 +60,7 @@ impl GroupedReduction for IsEmptyReduce {
         let chunks = values.chunks();
         assert!(chunks.len() == 1);
         let arr = &*chunks[0];
-        if arr.has_nulls() && !self.ignore_nulls {
+        if arr.has_nulls() && self.ignore_nulls {
             let valid = arr.validity().unwrap();
             for (i, g) in subset.iter().zip(group_idxs) {
                 let mut is_empty = self.is_empty.get_unchecked(g.idx());
@@ -68,7 +68,7 @@ impl GroupedReduction for IsEmptyReduce {
                     self.evicted_is_empty.push(is_empty);
                     is_empty = true;
                 }
-                is_empty |= valid.get_bit_unchecked(*i as usize);
+                is_empty &= !valid.get_bit_unchecked(*i as usize);
                 self.is_empty.set_unchecked(g.idx(), is_empty);
             }
         } else {

